core/modules/qproc/IndexMap.cc

```diff
@@ -0,0 +1,252 @@
+}
+template <>
+boost::shared_ptr<Region> make<Circle>(StringVector const& v) {
+    return lsst::qserv::qproc::getCircleFromParams(convertVec<Coordinate>(v));
+}
+template <>
+boost::shared_ptr<Region> make<Ellipse>(StringVector const& v) {
+    return lsst::qserv::qproc::getEllipseFromParams(convertVec<Coordinate>(v));
+}
+template <>
+boost::shared_ptr<Region> make<ConvexPolygon>(StringVector const& v) {
+    return lsst::qserv::qproc::getConvexPolyFromParams(convertVec<Coordinate>(v));
+}
+
+typedef boost::shared_ptr<Region>(*MakeFunc)(StringVector const& v);
+
+struct FuncMap {
+    FuncMap() {
+        fMap["box"] = make<Box>;
+        fMap["circle"] = make<Circle>;
+        fMap["ellipse"] = make<Ellipse>;
+        fMap["poly"] = make<ConvexPolygon>;
+        fMap["qserv_areaspec_box"] = make<Box>;
+        fMap["qserv_areaspec_circle"] = make<Circle>;
+        fMap["qserv_areaspec_ellipse"] = make<Ellipse>;
+        fMap["qserv_areaspec_poly"] = make<ConvexPolygon>;
+    }
+    typedef std::map<std::string, MakeFunc> Map;
+    Map fMap;
+};
+
+static FuncMap funcMap;
+} // anonymous namespace
+
+namespace lsst {
+namespace qserv {
+namespace qproc {
+typedef std::vector<boost::shared_ptr<Region> > RegionPtrVector;
+
+boost::shared_ptr<Region> getRegion(query::Constraint const& c) {
+    FuncMap::Map::const_iterator i = funcMap.fMap.find(c.name);
+    if(i != funcMap.fMap.end()) {
+        return i->second(c.params);
+    }
+    return boost::shared_ptr<Region>();
+}
+
+ChunkSpec convertSgSubChunks(SubChunks const& sc) {
+    ChunkSpec cs;
+    cs.chunkId = sc.chunkId;
+    cs.subChunks.resize(sc.subChunkIds.size());
+    std::copy(sc.subChunkIds.begin(), sc.subChunkIds.end(),
+              cs.subChunks.begin());
+    return cs;
+}
+
+bool isIndex(query::Constraint const& c) {
+    return c.name == "sIndex";
+}
+bool isNotIndex(query::Constraint const& c) {
+    return c.name != "sIndex";
+}
+
+////////////////////////////////////////////////////////////////////////
+// IndexMap::PartitioningMap definition and implementation
+////////////////////////////////////////////////////////////////////////
+class IndexMap::PartitioningMap {
+public:
+    explicit PartitioningMap(css::StripingParams const& sp) {
+        sg::Angle overlap(0.000001); // Dummy.
+        _chunker = boost::make_shared<sg::Chunker>(sp.stripes,
+                                                   sp.subStripes);
+
+    }
+    /// @return un-canonicalized vector<SubChunks> of concatenated region
+    /// results. Regions are assumed to be joined by implicit "OR" and not "AND"
+    ///
+    SubChunksVector getIntersect(RegionPtrVector const& rv) {
+        SubChunksVector scv;
+        for(RegionPtrVector::const_iterator i=rv.begin(), e=rv.end();
+            i != e;
+            ++i) {
+            if(*i) {
+                SubChunksVector area = getCoverage(**i);
+                scv.insert(scv.end(), area.begin(), area.end());
```

Comment:
Different area constraints linked by ORs can overlap, so that the same chunk and/or sub-chunks can appear in scv after this code runs. Do you get rid of duplicates somewhere else to avoid dispatching multiple times on the same chunk/sub-chunk?